backend/controllers/auth.controller.js

```diff
@@ -2,66 +2,123 @@ const User = require("../models/user.model");
 const bcrypt = require("bcrypt");
 const jwt = require("jsonwebtoken");
 const config = require("../config/config");
+const { commonErrorHandler } = require("../helper/errorHandler.helper");
+const { asyncHandler } = require("../middlewares/asyncHandler.middleware");
 
-const signUp = async (req, res) => {
-  const body = req.body;
-  const saltRounds = 10;
-  const salt = bcrypt.genSaltSync(saltRounds);
-  const hash = bcrypt.hashSync(body.password, salt);
-  body.password = hash;
-  const existingUser = await User.findOne({ email: body.email });
-  if (existingUser) {
-    return res.status(409).json({
-      error: "User already exists with this email id",
-    });
-  } else {
-    const user = new User(body);
-    user
-      .save()
-      .then((result) => {
-        return res.status(201).json({
-          message: "User registration successful",
-          data: result,
-        });
-      })
-      .catch((err) => {
-        return res.status(500).json({
-          error: "Server Error",
-        });
-      });
+// @desc    Signup user
+// @route   POST /api/auth/signup
+// @access  Public
+
+const userSignUp = asyncHandler(async (req, res) => {
+  try {
```

Comment:
you don't need try catch block for asyncHandler check and remove if its not required.